Source/CBLDatabase+Attachments.m

```diff
@@ -424,10 +432,16 @@ - (CBL_Revision*) updateAttachment: (NSString*)filename
     oldRev.properties = properties;
 
     // Store a new revision with the updated _attachments:
-    CBL_Revision* newRev = [self putRevision: oldRev prevRevisionID: oldRevID
-                             allowConflict: NO status: outStatus];
+    CBL_Revision* newRev = [self putRevision: oldRev prevRevisionID: oldRevID allowConflict: NO
+                                      status: outStatus error: outError];
     if (!body && *outStatus == kCBLStatusCreated)
         *outStatus = kCBLStatusOK;
+
+    if (CBLStatusIsError(*outStatus)) {
+        if (outError && !*outError)
```

Comment:
`*outError` is unknown at this point since we haven't initialized it — the caller didn't necessarily set it to nil. I think you can just take out `&& !*outError`.